src/vorta/views/partials/password_input.py

```diff
@@ -0,0 +1,137 @@
+from PyQt5.Qt import QObject
+from PyQt5.QtWidgets import QAction, QFormLayout, QLabel, QLineEdit, QWidget
+from vorta.i18n import translate
+from vorta.views.utils import get_colored_icon
+
+
+class PasswordLineEdit(QLineEdit):
+    def __init__(self, *, parent=None, show_visibility_button=True):
+        super().__init__(parent)
+
+        self._show_visibility_button = show_visibility_button
+        self._error_state = False
+        self._visible = False
+
+        self.setEchoMode(QLineEdit.Password)
+
+        if self._show_visibility_button:
+            self.showHideAction = QAction(self.tr("Show password"), self)
+            self.showHideAction.setCheckable(True)
+            self.showHideAction.toggled.connect(self.toggle_visibility)
+            self.showHideAction.setIcon(get_colored_icon("eye"))
+            self.addAction(self.showHideAction, QLineEdit.TrailingPosition)
+
+    def get_password(self):
+        """Return password text"""
+        return self.text()
+
+    def toggle_visibility(self):
+        """Toggle password visibility"""
+        self._visible = not self._visible
+        self.setEchoMode(QLineEdit.Normal if self._visible else QLineEdit.Password)
+
+        if self._show_visibility_button:
+            self.showHideAction.setIcon(get_colored_icon("eye"))
+        else:
+            self.showHideAction.setIcon(get_colored_icon("eye-slash"))
+
+    @property
+    def visible(self):
+        """Return password visibility"""
+        return self._visible
+
+    @property
+    def error_state(self):
+        """Return error state"""
+        return self._error_state
+
+    @error_state.setter
+    def error_state(self, error):
+        """Set error state and update style"""
+        self._error_state = error
+        if error:
+            self.setStyleSheet('border: 2px solid red;')
+        else:
+            self.setStyleSheet('')
+
+
+class PasswordInput(QObject):
+    def __init__(self, *, parent=None, minimum_length=9, show_error=True, label: list = None):
+        super().__init__(parent)
+        self._minimum_length = minimum_length
+        self._show_error = show_error
+
+        if label:
+            self._label_password = QLabel(label[0])
+            self._label_confirm = QLabel(label[1])
+        else:
+            self._label_password = QLabel(self.tr("Enter passphrase:"))
+            self._label_confirm = QLabel(self.tr("Confirm passphrase:"))
+
+        # Create password line edits
+        self.passwordLineEdit = PasswordLineEdit()
+        self.confirmLineEdit = PasswordLineEdit()
+        self.password_label = QLabel("")
```

Comment:
I think e.g. `validation_label` would fit better.

Reply:
What do you think?

Reply:
Yes, that sounds better.